go/vt/vttablet/tabletmanager/vreplication/vplayer.go

```diff
@@ -104,6 +112,37 @@ func newVPlayer(vr *vreplicator, settings binlogplayer.VRSettings, copyState map
 		settings.StopPos = pausePos
 		saveStop = false
 	}
+
+	queryFunc := func(ctx context.Context, sql string) (*sqltypes.Result, error) {
+		return vr.dbClient.ExecuteWithRetry(ctx, sql)
+	}
+	commitFunc := func(ctx context.Context) error {
+		return vr.dbClient.Commit()
+	}
+	batchMode := false
+	if vttablet.VReplicationExperimentalFlags&vttablet.VReplicationExperimentalFlagVPlayerBatching != 0 {
+		batchMode = true
+	}
+	if batchMode {
+		maxAllowedPacket := int64(1024) // 1KiB minimum in mysqld
+		res, err := vr.dbClient.ExecuteFetch("select @@session.max_allowed_packet as max_allowed_packet", 1)
+		if err != nil {
+			log.Errorf("Error getting max_allowed_packet, will use the minimum value: %v", err)
+		} else {
+			if maxAllowedPacket, err = res.Rows[0][0].ToInt64(); err != nil {
+				log.Errorf("Error getting max_allowed_packet, will use the minimum value:: %v", err)
+			}
+		}
+		maxAllowedPacket -= 512 // Leave 512 bytes of room for the vrepl record update, commit, etc.
```

Comment:
Yeah, it does. And I know that it can be large.... the column itself can be 10,000 bytes: https://github.com/vitessio/vitess/blob/main/go/vt/sidecardb/schema/vreplication/vreplication.sql#L22

At the same time, this isn't really a new problem -- we're actually paying attention to max allowed packet for the first time. max_allowed_packet (default is 64MiB in 8.0) can be smaller than the column... in which case things don't progress and you need to update the database config.

All that being said... I'll see if I can address this.

Reply:
I've properly incorporated that into the batch Mgmt here: https://github.com/vitessio/vitess/pull/14502/commits/706d7ddb196e3c1e4a3ad138b7db1e61d1211ead